core/shared/src/main/scala/zio/Schedule2.scala

```diff
@@ -0,0 +1,1009 @@
+   * A symbolic alias for `andThen`.
+   */
+  def ++[Env1 <: Env, In1 <: In, Out2 >: Out](that: Schedule2[Env1, In1, Out2]): Schedule2[Env1, In1, Out2] =
+    self andThen that
+
+  /**
+   * Returns a new schedule that allows choosing between feeding inputs to this schedule, or
+   * feeding inputs to the specified schedule.
+   */
+  def +++[Env1 <: Env, In2, Out2](
+    that: Schedule2[Env1, In2, Out2]
+  ): Schedule2[Env1, Either[In, In2], Either[Out, Out2]] = {
+    def loop(
+      self: StepFunction[Env, In, Out],
+      that: StepFunction[Env1, In2, Out2]
+    ): StepFunction[Env1, Either[In, In2], Either[Out, Out2]] =
+      (now: Instant, either: Either[In, In2]) => {
+        either match {
+          case Left(in) =>
+            self(now, in).map {
+              case Done(out)                     => Done(Left(out))
+              case Continue(out, interval, next) => Continue(Left(out), interval, loop(next, that))
+            }
+
+          case Right(in2) =>
+            that(now, in2).map {
+              case Done(out)                      => Done(Right(out))
+              case Continue(out2, interval, next) => Continue(Right(out2), interval, loop(self, next))
+            }
+        }
+      }
+
+    Schedule2(loop(self.step0, that.step0))
+  }
+
+  /**
+   * Operator alias for `andThenEither`.
+   */
+  def <||>[Env1 <: Env, In1 <: In, Out2](
+    that: Schedule2[Env1, In1, Out2]
+  ): Schedule2[Env1, In1, Either[Out, Out2]] = self.andThenEither(that)
+
+  /**
+   * The same as `&&`, but ignores the right output.
+   */
+  def <*[Env1 <: Env, In1 <: In, Out2](that: Schedule2[Env1, In1, Out2]): Schedule2[Env1, In1, Out] =
+    (self && that).map(_._1)
+
+  /**
+   * An operator alias for `zip`.
+   */
+  def <*>[Env1 <: Env, In1 <: In, Out2](that: Schedule2[Env1, In1, Out2]): Schedule2[Env1, In1, (Out, Out2)] =
+    self zip that
+
+  /**
+   * A backwards version of `>>>`.
+   */
+  def <<<[Env1 <: Env, In2](that: Schedule2[Env1, In2, In]): Schedule2[Env1, In2, Out] = that >>> self
+
+  /**
+   * Returns the composition of this schedule and the specified schedule, by piping the output of
+   * this one into the input of the other. Effects described by this schedule will always be
+   * executed before the effects described by the second schedule.
+   */
+  def >>>[Env1 <: Env, Out2](that: Schedule2[Env1, Out, Out2]): Schedule2[Env1, In, Out2] = {
+    def loop(self: StepFunction[Env, In, Out], that: StepFunction[Env1, Out, Out2]): StepFunction[Env1, In, Out2] =
+      (now: Instant, in: In) =>
+        self(now, in).flatMap {
+          case Done(out) => that(now, out).map(_.toDone)
+          case Continue(out, interval, next1) =>
+            that(now, out).map {
+              case Done(out2) => Done(out2)
+              case Continue(out2, interval2, next2) =>
+                val combined = (interval union interval2).getOrElse(interval min interval2)
+
+                Continue(out2, combined, loop(next1, next2))
```

Comment:
Loop returns right away, it returns a function, which isn't evaluated till later. So it's lazy recursion happening in the future.

Now every time we wrap a loop function, we add another level of indirection, and increase the stack frame size. So in essence the number of combinators you can apply to schedule is finite--too many, and you blow the stack. But we're already in that situation now since we already implement operators in existing schedule through "decoration".